x/gov/keeper/deposit.go

```diff
@@ -162,6 +162,44 @@ func (keeper Keeper) AddDeposit(ctx sdk.Context, proposalID uint64, depositorAdd
 	return activatedVotingPeriod, nil
 }
 
+// BurnAndSendDepositsToCommunityPool will burn the (deposits * prposal_cancel_burn_rate) amount of proposal deposits
+// and send remaining deposits of the proposal to community pool.
+func (keeper Keeper) BurnAndSendDepositsToCommunityPool(ctx sdk.Context, proposalID uint64, totalDeposits []sdk.Coin) error {
+	store := ctx.KVStore(keeper.storeKey)
+
+	proposalCancelBurnRate := keeper.GetParams(ctx).ProposalCancelBurnRate
+	// burn the deposits * proposal_cancel_burn_rate amount from proposal deposits (gov module)
+	burnRate := sdk.MustNewDecFromStr(proposalCancelBurnRate)
+
+	var burnDepositAmount sdk.Coins
+	for _, deposit := range totalDeposits {
+		burnAmount := sdk.NewCoin(
+			deposit.Denom,
+			sdk.NewDecFromInt(deposit.Amount).Mul(burnRate).RoundInt(),
+		)
+		burnDepositAmount = append(burnDepositAmount, burnAmount)
```

Comment:
Why are we calling `append` here? Since we're working with `Coins`, we should be calling `Add` instead.